test/instrumentation/modules/express.js

```diff
@@ -214,6 +214,45 @@ test('sub-routers include base path', function (t) {
   })
 })
 
+test('sub-routers throw exception', function (t) {
+  t.plan(6)
+
+  resetAgent(function (data) {
+    t.equal(data.transactions.length, 1, 'has a transaction')
+
+    var trans = data.transactions[0]
+    t.equal(trans.name, 'GET /api/:name', 'transaction name is GET /api/:name')
+    t.equal(trans.type, 'request', 'transaction type is request')
+  })
+
+  var error = new Error('hello')
+  var captureError = agent.captureError
+  agent.captureError = function (err, data) {
+    t.equal(err, error, 'has the expected error')
```

Comment:
For another question, I also reproduced locally in Express v4.4.5. My initial judgment is that the version did not throw the error exception parameter to the outer layer. 
But I still don't have a better solution to fix it except to comment this test code first. 

``` javascript
  t.plan(5)
  ......
  agent.captureError = function (err, data) {
    // t.equal(err, error, 'has the expected error')
    // t.ok(data, 'captured data with error')
  }
```
Can I fix this test error result like this?

Reply:
I just ran a test locally and it does seem like the error makes it to the root router. So I'm not sure why the agent doesn't pick it up... will just investigate a little more. But it might be an error unrelated to this.

Reply:
Ah, I found the issue. The agent expects the `finalhandler` module to be used by express so it can be notified of errors in case a custom error handler hasn't been manually added as middleware. But the `finalhandler` module wasn't added until Express 4.5.0, the version just after 4.4.5. That's why it's not calling `agent.captureError` in those early versions.

However, checking that the error is caught is not really something related to this test, so I suggest that you instead just change the custom `agent.captureError` function to be a no-op and always expect 4 assertions no matter which version of Express is being used.

Reply:
Thanks for your suggestion, I am modifying the code to a fixed 4 assertions.